src/test/java/org/example/services/ServicesTest.java

```diff
@@ -3,36 +3,181 @@
 import org.example.enums.*;
 import org.example.models.*;
 import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.*;
+import org.junit.jupiter.params.provider.*;
+
+import java.util.stream.*;
 
 class ServicesTest {
+    private final IServices service = new Services();
+
+    public static Stream<Arguments> provideArgsForScoreLessThanCost() {
+        return Stream.of(
+            Arguments.of(992, Weapons.KNIFE),
+            Arguments.of(9999, Weapons.SNIPER),
+            Arguments.of(1999, Weapons.KALASHNIKOV)
+        );
+    }
+
+    public static Stream<Arguments> provideArgsForScoreEqualOrMoreThanCost() {
+        return Stream.of(
+            Arguments.of(1001, Weapons.KNIFE),
+            Arguments.of(10_000, Weapons.SNIPER),
+            Arguments.of(20_543, Weapons.KALASHNIKOV)
+        );
+    }
 
-    @org.junit.jupiter.api.Test
-    void updateScoreService() {
+
+    @ParameterizedTest
+    @MethodSource("provideArgsForScoreLessThanCost")
+    void canBuyWeaponTest_ShouldReturnFalseWhenScoreIsLessThanCost(int cost, Weapons type) {
 
         //Given
-        int gave = 3;
-        int exp = 5;
+        WeaponModel weapon = new WeaponModel(type);
+
         //When
+        var result = service.canBuyWeapon(weapon, cost);
 
         //Then
-        Assertions.assertEquals(gave, exp);
+        Assertions.assertFalse(result);
+
     }
 
+    @ParameterizedTest
+    @MethodSource("provideArgsForScoreEqualOrMoreThanCost")
+    void canBuyWeaponTest_ShouldReturnTrueWhenScoreIsEqualOrBiggerThanCost() {
 
-    @org.junit.jupiter.api.Test
-    void canBuyWeaponTest_ShouldReturnFalse() {
+        //Given
+        WeaponModel weapon = new WeaponModel(Weapons.FIST);
+        int score = 100_000;
+
+        //When
+        var result = service.canBuyWeapon(weapon, score);
+
+        //Then
+        Assertions.assertTrue(result);
+
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {0, 6, 28, 496, 8128})
+    void isPerfectNumberTest_ShouldReturnTrueWhenPerfectNumber(int number) {
```

Comment:
Because the test fails for '0', either remove it or update the test to return the appropriate value... I think you need to refactor the logic of 'isPerfectNumber', it's not very clear/readable.